plugins/com.google.cloud.tools.eclipse.sdk/src/com/google/cloud/tools/eclipse/sdk/CloudSdkManager.java

```diff
@@ -42,4 +48,41 @@ public static boolean isManagedSdkFeatureEnabled() {
     }
     return false;
   }
+
+  /**
+   * Installs a Cloud SDK, if the preferences are configured to auto-manage the SDK. Blocks callers
+   * 1) if the managed SDK is being installed concurrently by others; and 2) until the installation
+   * is complete.
+   *
+   * @param consoleStream stream to which the install output is written
+   */
+  public static void installManagedSdk(MessageConsoleStream consoleStream)
+      throws CoreException, InterruptedException {
+    if (isManagedSdkFeatureEnabled()) {
+      if (CloudSdkPreferences.isAutoManaging()) {
+        runInstallJob(consoleStream, new CloudSdkInstallJob(consoleStream));
```

Comment:
I wondered if we could add an optimization to first check if the Managed Cloud SDK is installed before creating and running the job. But I suppose there's a change that an install-in-progress will appear to be present, resulting in using an incomplete install.  That's worth a comment here.

Reply:
Good idea.